backend/tracim_backend/lib/core/content.py

```diff
@@ -2378,3 +2378,43 @@ def create_todo(
         self.save(item, ActionDescription.CREATION, do_notify=do_notify)
 
         return item
+
+    def set_content_namespace(
+        self, content_id: int, content_namespace: ContentNamespaces
+    ) -> Content:
+        """Change the content namespace of a content
+
+        Args:
+            content_id: Id of the content
+            content_namespace: content_namespace wanted for the content
+
+        Returns:
+            Content: The modified content.
+        """
+        content = self.get_one(content_id=content_id)
+        if (
+            content.content_namespace != ContentNamespaces.PUBLICATION
+            or content_namespace != ContentNamespaces.CONTENT
+        ) and (
+            content.content_namespace != ContentNamespaces.CONTENT
+            or content_namespace != ContentNamespaces.PUBLICATION
+        ):
+            raise ContentNamespaceDoNotMatch(
+                "the content namespace wanted and the namespace of the content aren't allowed"
+            )
```

Comment:
This code part is difficult to read. If I understand correctly, the wanted behavior is to allow namespace transition only for PUBLICATION → CONTENT and CONTENT → PUBLICATION ? If I'm right, I think we can try to enhance readability of the code by something like : 

```python
is_publication_to_content = (
    content.content_namespace == ContentNamespaces.PUBLICATION
    and content_namespace == ContentNamespaces.CONTENT
)
is_content_to_publication = (
    content.content_namespace == ContentNamespaces.CONTENT
    and content_namespace == ContentNamespaces.PUBLICATION
)
if not any([is_publication_to_content, is_content_to_publication]):
            raise ContentNamespaceDoNotMatch(
                "the content namespace wanted and the namespace of the content aren't allowed"
            )
```

Reply:
done